component/src/main/java/org/wso2/extension/siddhi/store/redis/RedisIterator.java

```diff
@@ -0,0 +1,286 @@
+            }
+        }
+        if (scanResultsList.isEmpty() && stringCursor > 0) {
+            scanResultsList = fetchBatchOfResultsFromIndexTable(scanParams);
+            if (scanResultsList.isEmpty()) {
+                return resultList;
+            }
+        } else if (scanResultsList.isEmpty() && stringCursor == 0) {
+            return resultList;
+        }
+        scanResultsList.forEach(scanResult -> {
+            resultMap = jedis.hgetAll(scanResult.toString());
+            if (Objects.nonNull(resultMap)) {
+                result = resultsGenerator(resultMap);
+                resultList.add(result);
+            }
+        });
+        return resultList;
+    }
+
+    private List fetchBatchOfResultsFromIndexTable(ScanParams scanParams) {
+
+        scanResults = jedis.sscan(tableName + ":" + storeVariable.getName() + ":"
+                + streamVariable.getName(), String.valueOf(stringCursor), scanParams);
+        stringCursor = Long.valueOf(scanResults.getStringCursor());
+        return scanResults.getResult();
+    }
+
+    private List<Object[]> fetchAllResults() {
+        List<Object[]> resultList = new ArrayList<>();
+        if (isInitialTraverse) {
+            isInitialTraverse = false;
+            this.scanResultsList = fetchBatchOfResultsFromTable();
+            if (scanResultsList.isEmpty()) {
+                return resultList;
+            }
+        } else if (scanResultsList.isEmpty() && stringCursor > 0) {
+            this.scanResultsList = fetchBatchOfResultsFromTable();
+            if (scanResultsList.isEmpty()) {
+                return resultList;
+            }
+        } else if (scanResultsList.isEmpty() && stringCursor == 0) {
+            return resultList;
+        }
+
+        scanResultsList.forEach(e -> {
+            String type = jedis.type(e.toString());
+            if (type.equalsIgnoreCase("hash")) {
+                resultMap = jedis.hgetAll(e.toString());
+                if (Objects.nonNull(resultMap)) {
+                    result = resultsGenerator(resultMap);
+                    resultList.add(result);
+                }
+            }
+        });
+        return resultList;
+    }
+
+    private List fetchBatchOfResultsFromTable() {
+        ScanParams scanParams = new ScanParams();
+        scanParams.match(tableName + ":*");
+        scanParams.count(RedisTableConstants.REDIS_BATCH_SIZE);
+        scanResults = jedis.scan(String.valueOf(stringCursor), scanParams);
+        stringCursor = Long.valueOf(scanResults.getStringCursor());
+        return scanResults.getResult();
+    }
+
+
+    private Object[] resultsGenerator(Map<String, String> resultMap) {
+        Object[] generatedResult = new Object[attributes.size()];
+        int i = 0;
+        for (Attribute attribute : this.attributes) {
+            switch (attribute.getType()) {
+                case BOOL:
+                    generatedResult[i++] = Boolean.parseBoolean(resultMap.get(attribute.getName
```

Comment:
Please get this two lines into a single line

Reply:
Fixed by eef9f6d